src/Exceptions/BowlerGeneralException.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Vinelab\Bowler\Exceptions;
+
+use Exception;
+
+/**
+ * @author Kinane Domloje <kinane@vinelab.com>
+ */
+class BowlerGeneralException extends Exception
+{
+    protected $message;
+    protected $code;
+    protected $file;
+    protected $line;
+    protected $trace;
+    protected $previous;
+    protected $traceAsString;
+    protected $parameters;
+
+    // Dead lettering arguments
+    protected $arguments;
+
+    public function __construct($message, $code, $file, $line, $trace, $previous, $traceAsString, $parameters = [], $arguments = [])
+    {
+        $this->message = $message;
```

Comment:
use Parent as `parent::__construct($message, $code, $previous)` or simply pass `$e` as is and let the `Exception` do the work.